### PR TITLE
fix CI workflow

### PR DESCRIPTION
**Pull request by nam-jaehyun**
- fix CI workflow (including tests with Docker)
- match golang version to 1.18
- update vagrant setup
  - Ubuntu 21.10 is no longer supported; thus, we use Ubuntu 22.04 as NETNEXT=1

### CODE REVIEW

File: .github/workflows/ci-test-systemd.yml [hunk 1]
```diff
@@ -18,46 +18,59 @@ on:
...
+            for tool in "clang" "llc" "llvm-strip"; do
+              sudo rm -f /usr/bin/$tool
+              sudo ln -s /usr/bin/$tool-14 /usr/bin/$tool
+            done
+          fi
+
+      - name: Compile libbpf
+        run: |
+          cd KubeArmor/BPF
+          make -C libbpf/src
```
Comment (nyrahul) [inline .github/workflows/ci-test-systemd.yml:60]: Hey @nam-jaehyun should we keep this in script and invoke the script from all the actions? The script can be kept in `.github/scripts` folder.

**Review by nyrahul — CHANGES_REQUESTED** [review]
The GH action has substantial copy-pasta .. can we keep the llvm install handling as part of some script in .github folder and call it from the action?

**Review by daemon1024 — COMMENTED** [review]
LGTM, just as @nyrahul mentioned, if we can leverage [composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) or scripts. 

We can keep composite actions as a seperate work item/issue as well.

**Comment by nyrahul** [comment]
> LGTM, just as @nyrahul mentioned, if we can leverage [composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) or scripts.
> 
> We can keep composite actions as a seperate work item/issue as well.

I am going to just merge this and take up refactoring in separate PR.

**Review by nyrahul — APPROVED** [review]

**Pull request merged**